### PR TITLE
Update Redis app directory

### PR DESCRIPTION
**Pull request by razvand**
Rework the contents of the Redis app directory to make it more suitable for users (prime tool is KraftKit) and for developers (prime tools are Make, Kconfig, QEMU / Firecracker).

Move and output files in dedicated directories to keep the root directory clean: `kraft.yaml`, `README.md`, `Makefile`, `Makefile.uk` are the only files in the dedicated directories.

Update `README.md` according to the new structure.

Improve support for KraftKit: documentation, run scriipts, testing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,6 +55,47 @@ OK
...
+redis-qemu-x86_64-9pfs (qemu/x86_64)
+redis-qemu-x86_64-initrd (qemu/x86_64)
+```
+
+Note that, if you want to build for a different architecture then the previous build, currently you are required to remove the generated files.
+For that, run the command:
+
+```console
+rm -fr .config* .unikraft/
+```
```
Comment (StefanJum) [inline README.md]: Is that required when using `kraft`?

File: README.md [hunk 2]
```diff
@@ -173,41 +216,29 @@ Follow the steps below for the setup:
...
-
-     ```console
-     mkdir .unikraft
-     ```
-
-     Enter the `.unikraft/` directory:
-
-     ```console
-     cd .unikraft/
+     defconfigs/  fs0/  kraft.yaml  Makefile  Makefile.uk.default  README.md  run/  workdir/
```
Comment (StefanJum) [inline README.md]:
```suggestion
     defconfigs/  fs0/  kraft.yaml  Makefile  Makefile.uk  README.md  run/
```

File: README.md [hunk 3]
```diff
@@ -173,41 +216,29 @@ Follow the steps below for the setup:
...
+     git clone https://github.com/unikraft/lib-redis workdir/libs/redis
 
-     git clone https://github.com/unikraft/lib-musl libs/musl
+     git clone https://github.com/unikraft/lib-musl workdir/libs/musl
 
-     git clone https://github.com/unikraft/lib-lwip libs/lwip
+     git clone https://github.com/unikraft/lib-lwip workdir/libs/lwip
      ```
 
   1. Get back to the application directory:
```
Comment (StefanJum) [inline README.md]:
```suggestion
```
This is not needed anymore

**Review by StefanJum — CHANGES_REQUESTED** [review]
Looks good, just a few comments.

* Do we only provide basic config files? (i.e. not the `pie-paging` etc. ones)
* Would it make sense to keep the config naming convention kraft uses (`config.appname-arch...`)?

File: README.md (hunk 1 above)
Comment (razvand) [reply README.md]: Yes, I get errors if I don't do that.

**Comment by razvand** [comment]
> Do we only provide basic config files? (i.e. not the pie-paging etc. ones)

I will add all of them from [the `maintainer-tools` repository](https://github.com/unikraft-upb/maintainer-tools/tree/staging/workdir/copy/base/apps/redis).

> Would it make sense to keep the config naming convention kraft uses (`config.appname-arch...`)?

My thought was to simplify the names as much as possible. But if more people think we should use `config.appname-...`, then I will update accordingly. Cc: @skuenzer , @nderjung , @andreittr

File: README.md [hunk 4]
```diff
@@ -173,59 +216,41 @@ Follow the steps below for the setup:
       This will print the contents of the repository.
 
      ```text
-     fs0/  kraft.yaml  Makefile  Makefile.uk.default  README.md .config.redis-qemu-x86_64-9pfs .config.redis-qemu-aarch64-9pfs [...]  run-fc-x86_64-initrd.sh*  run-qemu-aarch64-9pfs.sh*  [...]
+     defconfigs/  fs0/  kraft.yaml  Makefile  Makefile.uk  README.md [...]  run-fc-x86_64-initrd.sh*  run-qemu-aarch64-9pfs.sh*  [...]
```
Comment (StefanJum) [inline README.md]:
```suggestion
     defconfigs/  fs0/  kraft.yaml  Makefile  Makefile.uk  README.md [...]  run/  [...]
```
These will be under the `run/` directory

**Review by StefanJum — COMMENTED** [review]
> I will add all of them from [the maintainer-tools repository](https://github.com/unikraft-upb/maintainer-tools/tree/staging/workdir/copy/base/apps/redis).

Perfect

> My thought was to simplify the names as much as possible. But if more people think we should use config.appname-..., then I will update accordingly. 

Makes sense, I don't have a strong opinion here, just if maybe the configs can be used by kraft somehow, but that should not be the case since kraft should only care about the `kraft.yaml` file.

It looks all good on my side besides the one small comment.

**Review by StefanJum — CHANGES_REQUESTED** [review]
Either add the `setup.sh` script (and mention it in the TLDR section) or add the cloning steps back into the `TLDR` section.

**Review by StefanJum — APPROVED** [review]
All good.
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>
Approved-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Pull request merged**